flamingo/scripts/hot_gas_plots.py

```diff
@@ -0,0 +1,460 @@
+            mask = np.where(n >= 1)    
+            
+            ax2.plot(bin_centers[mask], medians[mask], lw=2, label=leg)
+
+            # df_all = df_all.loc[df_all.in_FOF == True]
+            # X-ray Luminosity
+            inside = df_all.loc[df_all.distance <= 1]
+            groups = inside.groupby(['GroupNumber', 'm500'], as_index=False)
+
+            halos = groups.Lx.sum()
+            medians, bin_edges, binnumber = stats.binned_statistic(np.log10(halos.m500.values), np.log10(halos.Lx.values), statistic='median', bins=bin_edges)
+            n, bin_edges, binnumber = stats.binned_statistic(np.log10(halos.m500.values), np.log10(halos.Lx.values), statistic='count', bins=bin_edges)
+            per10, bin_edges, binnumber = stats.binned_statistic(np.log10(halos.m500.values), np.log10(halos.Lx.values), statistic=lambda y: np.percentile(y, 10), bins=bin_edges)
+            per90, bin_edges, binnumber = stats.binned_statistic(np.log10(halos.m500.values), np.log10(halos.Lx.values), statistic=lambda y: np.percentile(y, 90), bins=bin_edges)
+            mask = np.where(n >= 1)    
+
+            ax1.plot(bin_centers[mask], medians[mask], lw=2, alpha=1, zorder=10, label=leg)
+
+            # Density and Entropy profiles
+            
+            inside = df_all.loc[(df_all.m500 >= 5.25e13) & (df_all.m500 <= 2e14)]
+            # inside = df_all.loc[(df_all.m500 >= 5.75e13) & (df_all.m500 <= 1.3e14)]
+            # inside = df_all.loc[(df_all.m500 >= 6e13) & (df_all.m500 <= 1.17e14)]
+
+            print('median = %.2e' % inside.m500.median())
+            
+            #old
+            # inside = df_all.loc[(df_all.m500 >= 5.25e13) & (df_all.m500 <= 2e14)]
+            # inside = df_all.loc[(df_all.m500 >= 5.75e13) & (df_all.m500 <= 1.5e14)]
+                
+            full_index = np.arange(len(inside.GroupNumber.unique()))
+
+
+            df_rho = pd.DataFrame()
+            df_S = pd.DataFrame()
+            df_S_S500 = pd.DataFrame()
+            df_P = pd.DataFrame()
+            for j in range(len(centers)):
+                v_j = 'v_' + str(j)
+                df_i = inside.copy()
+                df_i = df_i.loc[(df_i.distance >= r_bins[j]) & (df_i.distance <= r_bins[j + 1])]
+                df_i['volume'] = (4./3) * np.pi * (((df_i.r500 * aexp * u.Mpc.to(u.cm) * r_bins[j + 1]) ** 3.) - ((df_i.r500 * aexp * u.Mpc.to(u.cm) * r_bins[j]) ** 3.))
+                df_i['rho_tot'] = df_i.mass / df_i.volume 
+                df_i['x_i'] = df_i.ne_nH / ((df_i.ne_nH + df_i.ni_nH) * df_i.mu * mp)
+                df_i['ne'] = df_i.rho_tot * df_i.x_i
+                df_i['wtemp'] = df_i.temp * df_i.mass
+                groups = df_i.groupby(['GroupNumber', 'm500'])
+                kT = kb / ergs_keV * groups.wtemp.sum() / groups.mass.sum()
+                p = kT * groups.ne.sum()
+                s = kT / groups.ne.sum() ** (2./3)
+                p /= groups.head(1).P500.median()
```

Comment:
@jemme07 I'm not sure I understand what you are doing here. Looks like you are constructing radial bins and then finding all halos that have particles within a bin. For those particles, you then compute the (total?) pressure and entropy within the bin and then you divide this by the median P500 and S500 of all halos that have particles in this particular bin. Is that right?